azkaban-common/src/main/resources/host_name_alias.perl

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env perl
+
+# A script to get the VIP name of the current host.
+
+use strict;
+use warnings;
+
+# Get a list of IP addresses that the host has.
+sub get_ip_addrs() {
+    my @ip_addrs = ();
+    # Use the absolute path of ifconfig
+    # since the script may be invoked without PATH being set.
+    for my $line (`/sbin/ifconfig -a`) {
+        # Parse the output of the ifconfig command
+        # looking for patterns of IP addresses
+        if ($line =~ /.*inet addr:(\d+\.\d+\.\d+\.\d+).*/) {
```

Comment:
IPv4 regex matching should bound the octets its only possible for 1-3 digits per octet.  For example: '\d{1,3}\.\d{1,3}\.\d{1,3}\.d{1,3}'  You can can condense this idea further but loose readability when having fun with regular expressions: "m!((\d{1,3}\.){3})(\d+)!" 

My other question, is  what happens if you get a ipv6 address? It's a possibility these days